assets/js/app.js

```diff
@@ -0,0 +1,206 @@
+const Player = (name, token) => {
+  const getPlayerName = () => name;
+  const getPlayerToken = () => token;
+
+  return { getPlayerName, getPlayerToken };
+};
+
+// Module Gameboard
+const gameBoard = (() => {
+  let board;
+  let currentPlayer;
+  let turnX;
+  let turnO;
+  let endRound;
+  let firstPlayer;
+  let secondPlayer;
+
+  const setGame = () => {
+    turnX = 'X';
+    turnO = 'O';
+    endRound = false;
+    board = [1, 2, 3, 4, 5, 6, 7, 8, 9];
+  };
+
+  const getCurrentPlayer = () => currentPlayer;
+
+  const winHorizontalCondition = () => {
+    if (
+      (board[0] === board[1] && board[0] === board[2])
+      || (board[3] === board[4] && board[3] === board[5])
+      || (board[6] === board[7] && board[6] === board[8])
+    ) {
+      return true;
+    }
+    return false;
+  };
+
+  const winVerticalCondition = () => {
+    if (
+      (board[0] === board[3] && board[0] === board[6])
+      || (board[1] === board[4] && board[1] === board[7])
+      || (board[2] === board[5] && board[2] === board[8])
+    ) {
+      return true;
+    }
+    return false;
+  };
+
+  const winDiagonalCondition = () => {
+    if (
+      (board[0] === board[4] && board[0] === board[8])
+      || (board[2] === board[4] && board[2] === board[6])
+    ) {
+      return true;
+    }
+    return false;
+  };
+
+  const players = (first, second) => {
+    firstPlayer = Player(first, turnX);
+    secondPlayer = Player(second, turnO);
+    currentPlayer = firstPlayer;
+  };
+
+  const changePlayer = () => {
+    currentPlayer = currentPlayer === firstPlayer ? secondPlayer : firstPlayer;
+    return currentPlayer;
+  };
+
+  const playerMove = (position) => {
+    if (endRound === false) {
+      if (currentPlayer === firstPlayer) {
+        board[position] = firstPlayer.getPlayerToken();
+        endRound = true;
+      } else {
+        board[position] = secondPlayer.getPlayerToken();
+        endRound = true;
+      }
+    }
+    changePlayer();
+    endRound = false;
+  };
+
+  const draw = () => {
+    const filter = board.filter(position => !Number.isInteger(position));
+    if (filter.length === 9) {
+      return true;
+    }
+    return false;
+  };
+
+  const win = () => {
+    if (
+      winDiagonalCondition()
+      || winHorizontalCondition()
+      || winVerticalCondition()
+    ) {
+      return true;
+    }
+    return false;
+  };
```

Comment:
This winning logic can be greatly simplified, Instead of creating three methods for every winning pattern, the player moves could be stored and then checked against a predefined  winning combination in an array, if that player moves exist in the array then the player won :thinking: :slightly_smiling_face: